base/linalg/triangular.jl

```diff
@@ -1894,15 +1884,20 @@ function sqrtm{T,realmatrix}(A::UpperTriangular{T},::Type{Val{realmatrix}})
     if realmatrix
         TT = typeof(sqrt(zero(T)))
     else
-        TT = typeof(sqrt(complex(-one(T))))
+        TT = typeof(sqrt(complex(zero(T))))
     end
-    n = checksquare(A)
+    n = Base.LinAlg.checksquare(A)
     R = zeros(TT, n, n)
-    for j = 1:n
-        R[j,j] = realmatrix ? sqrt(A[j,j]) : sqrt(complex(A[j,j]))
-        for i = j-1:-1:1
-            r = floop(A[i,j],R,i,j)
-            r==0 || (R[i,j] = r / (R[i,i] + R[j,j]))
+    @inbounds begin
+        for j = 1:n
+            R[j,j] = realmatrix ? sqrt(A[j,j]) : sqrt(complex(A[j,j]))
+            for i = j-1:-1:1
+                r = A[i,j] + zero(TT)
```

Comment:
Even simpler:
```jl
r::TT = A[i,j]
```
Though, like a lot of our generic linear-algebra code, this isn't right for dimensionful quantities, because `r` has the units of `A` whereas `R` has the units of `sqrt(A)`.  If we want to get that right, it should really be something like:
```
TA = realmatrix ? float(T) : Complex{float(T)}
```
and then use `r::TA`.